src/bibgetter.py

```diff
@@ -341,17 +506,177 @@ def format(filename):
         stdout=subprocess.DEVNULL,
     )
 
+def substitute_bibtex_key(entry_text, expected_key):
+    """
+    Substitute the record identifier of a bibtex entry.
+
+    We use this to print a bibtex entry with the record identifier that was requested by the user,
+    regardless of the record identifier used in the central bibliography.
+    """
+    lines = entry_text.splitlines()
+    # Extract entry id from first line
+    entry_id = re.search(r'{\s*([^,\s]+)', lines[0]).group(1)
+    lines[0] = lines[0].replace('{' + entry_id, '{' + expected_key)
+    # Replace expected_key with entry_id in ids field
+    for i, line in enumerate(lines):
+        if re.match(r'^\s*ids\s*=', line):
+            lines[i] = re.sub(rf'([^\w]|^){re.escape(expected_key)}([^\w]|$)', rf'\g<1>{entry_id}\g<2>', line)
+    return "\n".join(lines)
+
+def find_entry(key, central):
+    """
+    Returns the entry matching user-provided key, handling aliases and variants such as URLs
+    """
+    if not central:
+        return None
```

Comment:
One should explicitly document the behavior of this function when it returns `None`, because functions calling `find_entry` are supposed to deal with

* the central repository not being defined (is returning `None` then the right thing to do?)
* the entry not existing